### PR TITLE
docs: several fixes to docs of umbrella veneers for microgenerated clients

### PR DESCRIPTION
**Pull request by dazuma**
Covers all the wrapper veneers for current microgenerated libraries: `google-cloud-billing`, `google-cloud-recommender`, and `google-cloud-secret_manager`.

Specifics:
* Use the same `.gitignore` that is generated by the microgenerator
* Remove `.repo-metadata` entries that were used only to "generate" the authentication docs.
* Several fixes in the authentication docs including:
    * Fixed the client factory method calls (e.g. from `Google::Cloud::Billing.new` to `Google::Cloud::Billing.billing_service`)
    * Remove mentions of project ID environment variables which are not relevant to these libraries
    * Remove mention of the troubleshooting guide which doesn't exist for these libraries
* Remove some mentions of "langage" methods which were a copy-paste error.

### CODE REVIEW

File: google-cloud-billing/AUTHENTICATION.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ code.
 
 When running on Google Cloud Platform (GCP), including Google Compute Engine (GCE),
 Google Kubernetes Engine (GKE), Google App Engine (GAE), Google Cloud Functions
-(GCF) and Cloud Run, the **Project ID** and **Credentials** and are discovered
+(GCF) and Cloud Run, the **Credentials** and are discovered
```
Comment (quartzmo) [inline google-cloud-billing/AUTHENTICATION.md]: s/and are/are/

File: google-cloud-recommender/AUTHENTICATION.md [hunk 2]
```diff
@@ -49,7 +49,7 @@ code.
 
 When running on Google Cloud Platform (GCP), including Google Compute Engine (GCE),
 Google Kubernetes Engine (GKE), Google App Engine (GAE), Google Cloud Functions
-(GCF) and Cloud Run, the **Project ID** and **Credentials** and are discovered
+(GCF) and Cloud Run, the **Credentials** and are discovered
```
Comment (quartzmo) [inline google-cloud-recommender/AUTHENTICATION.md]: s/and are/are/

File: google-cloud-secret_manager/AUTHENTICATION.md [hunk 3]
```diff
@@ -49,7 +49,7 @@ code.
 
 When running on Google Cloud Platform (GCP), including Google Compute Engine (GCE),
 Google Kubernetes Engine (GKE), Google App Engine (GAE), Google Cloud Functions
-(GCF) and Cloud Run, the **Project ID** and **Credentials** and are discovered
+(GCF) and Cloud Run, the **Credentials** and are discovered
```
Comment (quartzmo) [inline google-cloud-secret_manager/AUTHENTICATION.md]: s/and are/are/

**Comment by dazuma** [comment]
@quartzmo Thanks for catching that. Fixed here. (That typo exists everywhere else too, so we'll also, separately, have to update the microgenerator and other places...)

**Pull request merged**